pkg/persistence/object_store.go

```diff
@@ -215,40 +218,28 @@ func (s *objectBackupStore) PutBackup(info BackupInfo) error {
 		return kerrors.NewAggregate([]error{err, deleteErr})
 	}
 
-	if err := seekAndPutObject(s.objectStore, s.bucket, s.layout.getPodVolumeBackupsKey(info.Name), info.PodVolumeBackups); err != nil {
-		errs := []error{err}
-
-		deleteErr := s.objectStore.DeleteObject(s.bucket, s.layout.getBackupContentsKey(info.Name))
-		errs = append(errs, deleteErr)
-
-		deleteErr = s.objectStore.DeleteObject(s.bucket, s.layout.getBackupMetadataKey(info.Name))
-		errs = append(errs, deleteErr)
-
-		return kerrors.NewAggregate(errs)
-	}
-
-	if err := seekAndPutObject(s.objectStore, s.bucket, s.layout.getBackupVolumeSnapshotsKey(info.Name), info.VolumeSnapshots); err != nil {
-		errs := []error{err}
-
-		deleteErr := s.objectStore.DeleteObject(s.bucket, s.layout.getBackupContentsKey(info.Name))
-		errs = append(errs, deleteErr)
-
-		deleteErr = s.objectStore.DeleteObject(s.bucket, s.layout.getBackupMetadataKey(info.Name))
-		errs = append(errs, deleteErr)
-
-		return kerrors.NewAggregate(errs)
+	// Since the logic for all of these files is the exact same except for the name and the contents,
+	// use a map literal to iterate through them and write them to the bucket.
+	var backupObjs = map[string]io.Reader{
+		s.layout.getPodVolumeBackupsKey(info.Name):       info.PodVolumeBackups,
+		s.layout.getBackupVolumeSnapshotsKey(info.Name):  info.VolumeSnapshots,
+		s.layout.getBackupResourceListKey(info.Name):     info.BackupResourceList,
+		s.layout.getCSIVolumeSnapshotKey(info.Name):      info.CSIVolumeSnapshots,
+		s.layout.getVolumeSnapshotContentsKey(info.Name): info.VolumeSnapshotContents,
 	}
 
-	if err := seekAndPutObject(s.objectStore, s.bucket, s.layout.getBackupResourceListKey(info.Name), info.BackupResourceList); err != nil {
-		errs := []error{err}
+	for key, reader := range backupObjs {
+		if err := seekAndPutObject(s.objectStore, s.bucket, key, reader); err != nil {
+			errs := []error{err}
 
-		deleteErr := s.objectStore.DeleteObject(s.bucket, s.layout.getBackupContentsKey(info.Name))
-		errs = append(errs, deleteErr)
+			// attempt to clean up the object in case it was made but we couldn't write contents.
+			deleteErr := s.objectStore.DeleteObject(s.bucket, key)
+			errs = append(errs, deleteErr)
 
-		deleteErr = s.objectStore.DeleteObject(s.bucket, s.layout.getBackupMetadataKey(info.Name))
-		errs = append(errs, deleteErr)
-
-		return kerrors.NewAggregate(errs)
+			deleteErr = s.objectStore.DeleteObject(s.bucket, s.layout.getBackupMetadataKey(info.Name))
```

Comment:
it looks like the old code deleted both the backup contents and the backup metadata here -- new code is missing the delete of the backup contents

Reply:
Ah, I see - I misread each of the blocks and what they were doing. I'll correct it.